store/tvShows.ts

```diff
@@ -0,0 +1,110 @@
+import { ActionTree, MutationTree } from 'vuex';
+import { BaseItemDto, BaseItemDtoQueryResult, ItemFields } from '~/api/models';
+
+export interface TvShowsState {
+  seasons: BaseItemDto[];
+  seasonEpisodes: BaseItemDto[][];
+}
+
+export const state = (): TvShowsState => ({
+  seasons: [],
+  seasonEpisodes: []
+});
+
+type MutationPayload = TvShowsState;
+
+export const mutations: MutationTree<TvShowsState> = {
+  ADD_TVSHOW_SEASONS(state: TvShowsState, { seasons }: MutationPayload) {
+    state.seasons.push(...seasons);
+  },
+  ADD_TVSHOW_SEASON_EPISODES(
+    state: TvShowsState,
+    { seasonEpisodes }: MutationPayload
+  ) {
+    state.seasonEpisodes.push(...seasonEpisodes);
+  },
+  CLEAR_TVSHOWS_SEASONS(state: TvShowsState) {
+    state.seasons = [];
+  }
+};
+
+export const enum TvShowsActions {
+  getTvShows = '[TV SHOWS] Get Seasons Request',
+  getTvShowsSuccess = '[TV SHOWS] Get Seasons Request Success',
+  getTvShowsFailure = '[TV SHOWS] Get Seasons Request Failure',
+  getTvShowsSeasonEpisodes = '[TV SHOWS] Get Season Episodes Request',
+  getTvShowsSeasonEpisodesSuccess = '[TV SHOWS] Get Season Episodes Request Success',
+  getTvShowsSeasonEpisodesFailure = '[TV SHOWS] Get Season Episodes Request Failure'
+}
+
+export const actions: ActionTree<TvShowsState, TvShowsState> = {
+  async [TvShowsActions.getTvShows]({ dispatch }, { item }) {
+    try {
+      const { data } = await this.$api.tvShows.getSeasons({
+        userId: this.$auth.user.Id,
+        seriesId: item.Id || ''
+      });
+
+      dispatch(TvShowsActions.getTvShowsSuccess, data);
+    } catch (err) {
+      dispatch(TvShowsActions.getTvShowsFailure, err);
+    }
+  },
```

Comment:
I agree with @MrTimscampi about this section, it seems a bit cryptic.

Reply:
Although stores in general are hard to read, so if Vue 3 will improve things it might be fine for now. Perhaps that library is worth the time investment.

Reply:
Well it really depends. I've used redux in all famous frameworks and to be honest `vuex` is at the bottom of my list for good redux usage inside a framework. All others provide type safety.